Strings refactor

OxrxL opened the pull request.

TheJeterLP commented:
All those string paths will have to be changed in the code too

OxrxL commented:
> All those string paths will have to be changed in the code too

Half in the settings xml (that doesn't exists yet on this repo?) of the app, and half on sponsorblock.

But shouldn't be a difficult task. I just replace any string name starting with "xfile_" or "vanced_" with "revanced_", and removed the useless stuff related to exoplayer v1 mod (that doesn't exista anymore on latest client versions).

TheJeterLP commented:
Well then please do that in the code too and then we can merge this PR

TheJeterLP commented:
> > All those string paths will have to be changed in the code too
> 
> Half in the settings xml (that doesn't exists yet on this repo?) of the app, and half on sponsorblock.
> 
> But shouldn't be a difficult task. I just replace any string name starting with "xfile_" or "vanced_" with "revanced_", and removed the useless stuff related to exoplayer v1 mod (that doesn't exists anymore on latest app versions).

There is no settings XML. String paths are located in the java code, most of them in Sponsorblock and settings package.

OxrxL commented:
> > > All those string paths will have to be changed in the code too
> > 
> > 
> > Half in the settings xml (that doesn't exists yet on this repo?) of the app, and half on sponsorblock.
> > But shouldn't be a difficult task. I just replace any string name starting with "xfile_" or "vanced_" with "revanced_", and removed the useless stuff related to exoplayer v1 mod (that doesn't exists anymore on latest app versions).
> 
> There is no settings XML. String paths are located in the java code, most of them in Sponsorblock and settings package.

Without an xml link, custom settings can't be renderized by the app.

On last vanced version (17.03.38) there is a file named "xfile_settings" where all these strings are placed.

It's the same concept as the "microG" option.

TheJeterLP commented:
I'll have a look into that later

OxrxL commented:
> I'll have a look into that later

Little typo: xfile_prefs.xml

<img width="960" alt="Immagine 2022-06-30 064935" src="https://user-images.githubusercontent.com/108184954/176595265-9becd38c-19e7-4ddf-9005-56d54e344759.png">

OxrxL commented:
Further links are placed on settings_fragment.xml

<img width="960" alt="test" src="https://user-images.githubusercontent.com/108184954/176595480-3d549bed-0dad-4448-88d7-df0ad513cc3d.png">

TheJeterLP commented:
Anyways, please change the string paths that are located in the java code too

TheJeterLP commented:
> > I'll have a look into that later
> 
> Little typo: xfile_prefs.xml
> 
> <img width="960" alt="Immagine 2022-06-30 064935" src="https://user-images.githubusercontent.com/108184954/176595265-9becd38c-19e7-4ddf-9005-56d54e344759.png">
> 

Vanced team probably created/edited those files by hand, so we will have to create a patch for that once the patcher is able to merge resources to revanced

OxrxL commented:
@TheJeterLP I added xml files (adapted to revanced) too.

Give it a look.

P.S: Obviously settings fragment will not have to be overwritten, but the content of the file I added in this PR will be added to that of the original file in the apk

OxrxL commented on app/src/main/java/app/revanced/integrations/settingsmenu/ReVancedSettingsFragment.java:
```diff
@@ -303,10 +303,10 @@ public void onCreate(Bundle bundle) {
             this.settingsInitialized = true;
         } catch (Throwable th) {
-            LogHelper.printException("XSettingsFragment", "Unable to retrieve resourceId for xfile_prefs", th);
+            LogHelper.printException("ReVancedSettingsFragment", "Unable to retrieve resourceId for revanced_prefs", th);
```
It's already renamed.

OxrxL commented on app/src/main/java/app/revanced/integrations/settings/SettingsEnum.java:
```diff
@@ -8,10 +8,10 @@
     MANUFACTURER_OVERRIDE_STRING("override_manufacturer", null),
     MODEL_OVERRIDE_STRING("override_model", null),
-    CODEC_OVERRIDE_BOOLEAN("override_resolution_xfile_enabled", true),
+    CODEC_OVERRIDE_BOOLEAN("override_resolution_revanced_enabled", true),
```
This can't be done without a further refactor. Otherwise it generate conflicts (there are already strings named like this).

TheJeterLP replied:
Then rename those other strings please

TheJeterLP commented:
Thanks!
